drf_kit/views/viewsets.py

```diff
@@ -156,7 +158,16 @@ class SearchMixin:
         filter_backends=[filters.FilterInBodyBackend, *api_settings.DEFAULT_FILTER_BACKENDS],
     )
     def search(self, request):
-        return self.list(request)
+        #  Before making the proxy to `list`, convert data from body to query params
+        new_request = clone_request(request=request, method="GET")
```

Comment:
It looks fishy to clone the request and change its attributes "just" to make cache work.
What if someone attempts to check if it's a POST down the road? It won't work. It might even make it harder to debug.

A better approach could be supporting cache directly on the [cache feature](https://github.com/flamingo-run/drf-toolkit/blob/main/drf_kit/cache.py): detect if it's a `POST /search` and them allow the cache to be built using the body.

Reply:
I was thinking about this. I don't like the approach of inspecting the request inside cache feature to check whether that request came from a `/search` call, cause it mixes concerns from different domains into cache.

Wdyt of making cache aware of both `body` and `query_params`? That way cache does not know about `/search` and vice-versa. Something like this:

```
class QueryListParamsKeyBit(bits.AllArgsMixin, bits.KeyBitDictBase):
    def get_source_dict(self, params, view_instance, view_method, request, args, kwargs):
        data = request.data | {k: request.query_params.getlist(k) for k in request.GET}
        data["renderer_type"] = request.accepted_media_type
        return data
```

Instead of this:

```
class QueryListParamsKeyBit(bits.AllArgsMixin, bits.KeyBitDictBase):
    def get_source_dict(self, params, view_instance, view_method, request, args, kwargs):
        data = {k: request.query_params.getlist(k) for k in request.GET}
        data["renderer_type"] = request.accepted_media_type
        return data
```

Change on `data = {k: request.query_params.getlist(k) for k in request.GET}` line only. The first one has `request.data` as the initial dict.

Reply:
Another approach could be:
- create a custom a cache-key constructor for the /search
- set it as decorator ([example](https://chibisov.github.io/drf-extensions/docs/#cache-key))

What do you think?

Reply:
Sounds great and straightforward. Adding this